From 37a20df4e8e1c534687eb5489130d1007796e5cf Mon Sep 17 00:00:00 2001
From: Tatsuo ONISHI <onishi@latmos.ipsl.fr>
Date: Wed, 23 Nov 2022 14:48:54 +0100
Subject: [PATCH 34/50] (Update) ph_cmuaq_cur initialization (for -D option)

TYPE : Update

Description:
       ph_cmuaq_cur is a local variable in chem/module_mosaic_cloudchem.F
       But it is not initialized and causes an error when compiled with -D option

List of modified files:
   * chem/module_mosaic_cloudchem.F
---
 chem/module_mosaic_cloudchem.F | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/chem/module_mosaic_cloudchem.F b/chem/module_mosaic_cloudchem.F
index 4413b029..d4fa30b3 100644
--- a/chem/module_mosaic_cloudchem.F
+++ b/chem/module_mosaic_cloudchem.F
@@ -343,8 +343,12 @@
 #endif
 
 	gas_aqfrac_box(:) = 0.0
-
-
+!TO2
+    if (isnan(ph_cmuaq_cur)) then
+      ph_cmuaq_cur = 0.0
+    endif
+!    write(*,*) 'before interface_to_aqoperator1, ph_cmuaq_cur = ',ph_cmuaq_cur
+!TO2
 !   make call to interface_to_aqoperator1
 	call interface_to_aqoperator1(   &
 	    istat_aqop,   &
@@ -718,6 +722,7 @@
 	xprescribe_ph_dp = 0.0d0
 	if (xprescribe_ph .ne. 0.0) xprescribe_ph_dp = xprescribe_ph
 	ph_cmuaq_cur_dp = 0.0d0
+!        write(*,*) 'ph_cmuaq_cur = ',ph_cmuaq_cur
 	if (ph_cmuaq_cur .ne. 0.0) ph_cmuaq_cur_dp = ph_cmuaq_cur
 
 	do i = 1, ngas
-- 
2.31.1

